File: tracking-program/8087/api/models/MeterCSV.js
/**
 * MeterCSV.js
 *
 * @description :: A stored meter data CSV.
 */

module.exports = {

  attributes: {

    // The type of meter CSV report that this record represents.
    // Use a constant from the `sails.config.constants.METER_CSV_TYPES` set.
    reportType: {
      type: 'number',
      required: true
    },

    // The title of the saved CSV report.
    title: {
      type: 'string',
      required: true
    },

    // The UUID of the saved report, used for retrieving a URL from S3.
    uuid: {
      type: 'string',
      required: true
    },

    // The project under which this report was generated.
    project: {
      model: 'project',
      required: true
    },

    // The earliest date represented in the report data.
    fromDate: {
      type: 'number',
      required: true
    },

    // The latest date represented in the report data.
    toDate: {
      type: 'number',
      required: true
    },

    // The power meters represented in the report data.
    meters: {
      collection: 'meter'
    },

    // The users who were emailed the report after it was generated.
    users: {
      collection: 'user'
    }

  },

};

File: tracking-program/8087/flask_app/app/models/meter_csv.py
"""MeterCSV model - api/models/MeterCSV.js"""
from app.extensions import db
from app.models.base import BaseModel


class MeterCSV(BaseModel):
    __tablename__ = "metercsv"

    reportType = db.Column(db.Integer, nullable=False)
    title = db.Column(db.String(255), nullable=False)
    uuid = db.Column(db.String(255), nullable=False)
    project = db.Column(db.Integer, db.ForeignKey("project.id"), nullable=False)
    fromDate = db.Column(db.BigInteger, nullable=False)
    toDate = db.Column(db.BigInteger, nullable=False)
